stackrox/stackrox
# ROX-22052: Add discovered clusters service on frontend

## Description

Frontend counterpart to #9736 and #9769

DiscoveredClusterService.ts file corresponds to
https://github.com/stackrox/stackrox/blob/master/proto/api/v1/discovered_cluster_service.proto

1. Add src/Containers/Clusters/DiscoveredClusters/DiscoveredCluster.tsx
    * Add helper functions for icon and text.
2. Edit src/Containers/Clusters/DiscoveredClusters/DiscoveredClustersPage.tsx
    * Uncomment search filter.
3. Edit src/Containers/Clusters/DiscoveredClusters/DiscoveredClustersToolbar.tsx
4. Edit src/Containers/Clusters/DiscoveredClusters/DiscoveredClustersTable.tsx
    * Call helper functions for icon and text.
5. Edit src/Containers/Clusters/DiscoveredClusters/DiscoveredClustersToolbar.tsx
    * Add `setTypesSelected` function and commented out functions.
    * Add comments for additional filter elements.
    * Update props of `SearchFilterType` element.
6. Edit src/Containers/Clusters/DiscoveredClusters/SearchFilterType.tsx
    * Render types from service and call helper function for text.
    * Update `searchFilter` for temporary single select.
7. Rename src/services/DiscoveredClustersService (plural) as DiscoveredClusterService (singular) for consistency with backend file name
    * Add request and response types.
    * Add service functions.
    * Add filter functions.

### Residue

1. Add `SearchFilterName` element.
2. Add `SearchFilterStatuses` element
3. Replace `SearchFilterType` (singular) with `SearchFilterTypes` (multiple) element.

## Checklist

## Testing Performed

Before you `yarn deploy` in ui folder:

```sh
export ROX_CLOUD_SOURCES=true
```

1. `yarn lint` in ui
2. `yarn build` in ui and then compute branch - master for the following
    * `wc build/static/js/*.js`
        main.js 0 = 3947282 - 3947282
        total 4625 = 10510661 - 10506036
    * `ls -al build/static/js/*.js | wc`
        files 0 = 92 - 92
3. `yarn start` in ui

### Manual testing

1. Visit /main/clusters/discovered-clusters
    * See absence of query string in page address
    * GET /v1/count/discovered-clusters?
    * See GET /v1/discovered-clusters?pagination.limit=10&pagination.offset=0&pagination.sortOption.field=Cluster&pagination.sortOption.reversed=false request
    * See **All types** in search filter element
    * See **No discovered clusters**
    ![query_none](https://github.com/stackrox/stackrox/assets/11862657/0d58004c-9647-4fe2-86f5-223cbeb00b51)

2. Select **EKS**
    * See `?s[Cluster%20Type][0]=EKS` in page address
    * GET /v1/count/discovered-clusters?filter.types=EKS
    * GET /v1/discovered-clusters?filter.types=EKS&pagination.limit=10&pagination.offset=0&pagination.sortOption.field=Cluster&pagination.sortOption.reversed=false
    * See **EKS** in search filter element
    * See **No discovered clusters found**
    ![query_types_EKS](https://github.com/stackrox/stackrox/assets/11862657/05f10bc8-e66f-4c05-848d-29d2ce4bad77)

3. Select **All types**
    * See same as step 1 above
